.github/workflows/build-argocd-controller.yaml

```diff
@@ -0,0 +1,155 @@
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v3
+        with:
+          version: 'latest'
+
+      - name: Set up Kubernetes
+        uses: azure/setup-kubectl@v4
+
+      - name: Install yq
+        run: |
+          wget https://github.com/mikefarah/yq/releases/download/v4.35.1/yq_linux_amd64 -O /usr/local/bin/yq
+          chmod +x /usr/local/bin/yq
+
+      - name: Install up CLI
+        run: |
+          curl -sL "https://cli.upbound.io" | sh
+          sudo mv up /usr/local/bin/
+          up version
+
+      - name: Create working directory
+        run: |
+          mkdir -p controller-package
+          cd controller-package
+
+      - name: Set version variables
+        run: |
+          echo "ARGO_VERSION=${{ github.event.inputs.argo_version }}" >> $GITHUB_ENV
+          echo "CONTROLLER_VERSION=${{ github.event.inputs.controller_version }}" >> $GITHUB_ENV
+          echo "PUSH_TO_REGISTRY=${{ github.event.inputs.push_to_registry }}" >> $GITHUB_ENV
+
+      - name: Pull Helm chart
+        working-directory: controller-package
+        run: |
+          helm pull ${{ env.CHART_NAME }} --repo ${{ env.CHART_REPOSITORY }} --version ${{ env.ARGO_VERSION }}
+
+      - name: Setup controller package structure
+        working-directory: controller-package
+        run: |
+          # Create helm directory and move chart
+          mkdir -p helm
+          mv ${{ env.CHART_NAME }}-${{ env.ARGO_VERSION }}.tgz helm/chart.tgz
+
+          # Extract CRDs
+          mkdir -p crds
+          helm template ${{ env.RELEASE_NAME }} helm/chart.tgz -n ${{ env.RELEASE_NAMESPACE }} --include-crds --kube-version=1.25.0 | \
+            yq e 'select(.kind == "CustomResourceDefinition")' - | \
+            yq -s '("crds/" + .metadata.name + ".yaml")' -
+
+          # Create crossplane.yaml
+          cat <<EOF > crossplane.yaml
+          apiVersion: meta.pkg.upbound.io/v1alpha1
+          kind: Controller
+          metadata:
+            annotations:
+              friendly-name.meta.crossplane.io: Controller ArgoCD
+              meta.crossplane.io/description: |
+                The ArgoCD Controller enables continuous delivery and declarative configuration
+                management for Kubernetes applications using GitOps principles.
+              meta.crossplane.io/license: Apache-2.0
+              meta.crossplane.io/maintainer: Upbound Maintainers <info@upbound.io>
+              meta.crossplane.io/readme: |
+                ArgoCD is a declarative GitOps continuous delivery tool for Kubernetes that
+                follows the GitOps methodology to manage infrastructure and application
+                configurations.
+              meta.crossplane.io/source: https://github.com/argoproj/argo-cd
+            name: argocd
+          spec:
+            packagingType: Helm
+            helm:
+              releaseName: ${{ env.RELEASE_NAME }}
+              releaseNamespace: ${{ env.RELEASE_NAMESPACE }}
```

Comment:
We will only support ArgoCD core as a controller: https://argo-cd.readthedocs.io/en/stable/operator-manual/core/
There needs to be some value overrides here for that.

Reply:
we working on this - it's more a prereq. this PR

Reply:
xref: https://github.com/argoproj/argo-helm/issues/1823